### ISSUE TITLE
Arrow functions: "Parsing error: unexpected token ="

### ISSUE THREAD

**Issue by mallorybulkley** [eslint/eslint#10137]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 7.9.0
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?** 
```  "babel-eslint": "^8.2.2",```

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```js
"eslint": "^4.19.1",
"eslint-config-airbnb": "^16.1.0",
"eslint-plugin-import": "^2.9.0",
"eslint-plugin-jest": "^21.15.0",
"eslint-plugin-jsx-a11y": "^6.0.3",
"eslint-plugin-react": "^7.7.0",
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
anyFunc = (e) => {
    // does not matter what is inside
  }
```

```bash
./node_modules/.bin/eslint myfile.jsx
```

**What did you expect to happen?**
Errors should be output and shown in VS Code. 

**What actually happened? Please include the actual, raw output from ESLint.**
```
34:16 error  Parsing error: Unexpected token =

✖ 1 problem (1 error, 0 warnings)
```
VS Code integration breaks: the first = is underlined with this message:`[eslint] Parsing error: unexpected token =`. Linting is broken for entire file, no other errors are shown even though they exist, and autofix on save does not work.

Everything works as expected for files without arrow functions both via CL and in VSCode.

**Comment by gyandeeps** [comment]
Can you share you eslint config?

**Comment by mallorybulkley** [comment]
```
extends: airbnb

plugins:
  - react
  - jest
  
globals:
  __DEBUG_SERVER_ERRORS__: true
  __SERVER_ERRORS__: true
  fabric: true

env:
  browser: true
  node: true
  jest/globals: true
```

**Comment by mallorybulkley** [comment]
Mostly just specifics about globals and rules, that all comes from airbnb, nothing unexpected. Here's the rest.
```
"parserOptions": {
    "ecmaFeatures": {
      "globalReturn": true,
      "generators": false,
      "objectLiteralDuplicateProperties": false,
      "experimentalObjectRestSpread": true,
      "jsx": true
    },
    "ecmaVersion": 2017,
    "sourceType": "module"
  },
"plugins": [
    "import",
    "react",
    "jsx-a11y",
    "jest"
  ],
  "settings": {
    "import/resolver": {
      "node": {
        "extensions": [
          ".js",
          ".jsx",
          ".json"
        ]
      }
    },
    "import/extensions": [
      ".js",
      ".jsx"
    ],
    "import/core-modules": [],
    "import/ignore": [
      "node_modules",
      "\\.(coffee|scss|css|less|hbs|svg|json)$"
    ],
    "react": {
      "pragma": "React",
      "version": "15.0"
    },
    "propWrapperFunctions": [
      "forbidExtraProps",
      "exact",
      "Object.freeze"
    ]
  },
```

**Comment by euharrison** [comment]
You should add `babel-eslint`.

```yml
extends: airbnb
parser: babel-eslint
plugins:
  - react
  - jest
(...)
```

**Comment by slorenzo** [comment]
@euharrison Thanks! Adding `babel-eslint` solves the issue.

**Comment by mallorybulkley** [comment]
Yes - adding the parser worked. Thanks!

**Comment by tanyta78** [comment]
@euharrison Thanks! Adding babel-eslint solves the issue.

**Comment by ChampIsMe** [comment]
How to add babel-eslint was not indicated

**Comment by Dynogic** [comment]
Thanks!

**Comment by ViggoV** [comment]
For some reason my locally installed version of eslint seem to behave differently from my globally installed version, both `5.9.0`. Without `babel-eslint` the local version will throw an `unexpected token =` error for `=`-signs in jsx attributes.. `babel-eslint`, however, did the trick.

**Comment by m3t4lch7** [comment]
@ChampIsMe  
> How to add babel-eslint was not indicated

It kinda was, but just not in the json format.
Open the `.eslintrc.json` file and add: `"parser": "babel-eslint",`
